Docs: Adapt the Diataxis template and restructure the content for Basic Authentication (#27105)
quarkusio/quarkus: docs/src/main/asciidoc/security-basic-auth-concept.adoc

[id="security-basic-auth-concept"]
= Basic Authentication
 

> HTTP Basic Authentication uses fields in the HTTP header and is the easiest to set up. Also, it is one of the least resource-demanding techniques that enforce access controls to the Web resources without requiring HTTP cookies, session identifiers, or login pages.

Reviewer: ```suggestion
HTTP Basic Authentication is one of the least resource-demanding techniques that enforce access controls to the Web resources.
It uses fields in the HTTP header and does not require HTTP cookies, session identifiers, or login pages. 
```

Reviewer: Single sentence per line..

Author: Done

> In the context of an HTTP request, Basic authentication is a method for an HTTP user agent, such as a web browser, to provide a user name and password when creating a request. In Basic HTTP Authentication, a request contains a header field in the form of `Authorization: Basic <credentials>`, where credentials are the Base64 encoding of a user ID and password joined by a colon as described in the following example.

Reviewer: ```suggestion
An HTTP user agent, such as a web browser, uses an   `Authorization` header to provide a user name and password in each HTTP request.
The header is specified as `Authorization: Basic <credentials>`, where credentials are the Base64 encoding of the user ID and password joined by a colon, as shown in the following example.
```

One line per sentence. This could be differently combined with the previous grouping... ?

Author: Done

====
.Example
 

> If the user name is `Alice` and the password is `secret`, the HTTP authorization header looks authorization as `Authorization: Basic  QWxjZTpzZWNyZXQ=`, where `QWxjZTpzZWNyZXQ=` is a Base64 encoded representation of the `Alice:secret` string.

Reviewer: ```suggestion
If the user name is `Alice` and the password is `secret`, the HTTP authorization header would be `Authorization: Basic  QWxjZTpzZWNyZXQ=`, where `QWxjZTpzZWNyZXQ=` is a Base64 encoded representation of the `Alice:secret` string.
```

Author: How about keeping it as is and using the simple present tense?

Reviewer 2: Will reflect this in my review. Thank you! :)

====
 

> The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.

Reviewer: ```suggestion
The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. 
The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. 
Basic Authentication should be used with HTTPS to provide confidentiality.
```

One line per sentence. People can use Basic Auth without HTTPS, something that is mentioned again in the section below.

Perhaps this paragraph should be combined with the first section of the second? Not sure.

Reviewer 2: Why we cant write 2 sentences per line, if the context is the same and we dont want to create a lot of white space?

Reviewer: It doesn't leave whitespace or create a break in the paragraph. 
It does help with diffs over time when reviewing edits.

https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/doc-reference.adoc#semantic-line-breaks

https://rhodesmill.org/brandon/2012/one-sentence-per-line/

Author: Changed to one sentence per line.

> Basic Authentication is a well-specified, simple challenge and response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:

Reviewer: ```suggestion
Basic Authentication is a well-specified, simple challenge and response scheme that all web browsers and most web servers understand. 
However, there are a few limitations associated with Basic Authentication, which include:
```

Author: Done

Credentials are sent as plain text::
+
--

> It is required to use HTTPS with Basic Authentication to avoid exposing the credentials. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when a request is forwarded to Quarkus over HTTP.

Reviewer: ```suggestion
Use HTTPS with Basic Authentication to avoid exposing the credentials. 
The risk of exposing credentials as plain text increases if a load balancer terminates HTTPS, as the request is then forwarded to Quarkus over HTTP.
```

Author: Done

> Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only, and the credentials are propagated to the next Quarkus endpoint over HTTP.

Reviewer: This should be combined with the previous statement somehow. Load-balancer or multi-hop HTTPS (or SSL) termination is functionally the same.

Reviewer: I don't see how this comment has been resolved? I would like to see something more like: 

Use HTTPS with Basic Authentication to avoid exposing the credentials. The risk of exposing credentials increases if the HTTPS connection is terminated by either a load balancer or a routing service in a multi-hop deployment, as the request, including the credentials stored in the header, is forwarded to Quarkus in plain text over an unencrypted HTTP connection.

--

Credentials are sent with each request::
+
--
In Basic Authentication, a username and password need to be sent with each request, which increases the risk of credentials being exposed.
--

Application complexity increases::
+
--

> The Quarkus application needs to ensure itself that usernames, passwords, and roles are managed securely, which increases the application complexity.

Reviewer: ```suggestion
The Quarkus application itself needs to validate that   usernames, passwords, and roles are managed securely.
This can introduce significant complexity to the application. 
Depending on the use case, other authentication mechanisms that delegate user, password, and role management to specialized services may be a better choice.
```

Author: Done

Application complexity increases::
+
--
The Quarkus application must validate that usernames, passwords, and roles are managed securely.
This process, however, can introduce significant complexity to the application.

> Depending on the use case, other authentication mechanisms delegate username, password, and role management to specialized services.

Reviewer: ```suggestion
Depending on the use case, other authentication mechanisms that delegate username, password, and role management to specialized services may be a better choice.
```

Author: Done
